Dynamic script icons (#1)
AprilSylph/XKit-Rewritten: docs/framework.md

## Metadata files

> Each module should be accompanied by a `.json` file of matching name; i.e. `example.js` should be accompanied by `example.json` in the same directory level. Right now, only five keys are supported; expect more to be added in future.

Reviewer: ```suggestion
Each module should be accompanied by a `.json` file of matching name, i.e. `example.js` should be accompanied by `example.json` in the same directory level. Right now, four keys are supported.
```
(i know the `"preferences"` documentation is missing but the number may as well be accurate in the meantime)
